contracts/CrossChain/SgInfo.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.14;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/CrossChain/ISgInfo.sol";
+
+/// @title cross chain router implementation, version 3.
+/// @notice Router for stateless execution of cross chain swap against WOOFi or 1inch swap.
+/// @custom:stargate-contracts https://stargateprotocol.gitbook.io/stargate/developers/contract-addresses/mainnet
+contract SgInfo is ISgInfo, Ownable {
+    mapping(uint16 => address) public sgETHs; // chainId => SGETH token address
+
+    mapping(uint16 => mapping(address => uint256)) public sgPoolIds; // chainId => token address => Stargate poolId
+
+    address public stargateRouter;
+    uint16 public sgChainIdLocal; // Stargate chainId on local chain
+
+    constructor(address _stargateRouter, uint16 _sgChainIdLocal) {
+        _initSgETHs();
+        _initSgPoolIds();
+        stargateRouter = _stargateRouter;
+        sgChainIdLocal = _sgChainIdLocal;
```

Comment:
Better to be moving this part to the beginning

Reply:
resolved